pkg/codegen/dotnet/utilities.go

```diff
@@ -118,3 +119,23 @@ func makeSafeEnumName(name, typeName string) (string, error) {
 
 	return safeName, nil
 }
+
+// Provides code for a method which will be placed in the program preamble if deemed
+// necessary. Because many Terraform functions are complex, it is much prettier to
+// encapsulate them as their own function in the preamble.
+func getHelperMethodIfNeeded(functionName string) (string, bool) {
+	var methodBody string
+
+	switch functionName {
+	case "sha1":
+		methodBody =
+			`private static string sha1(string input) {
```

Comment:
The [convention](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions) in .NET is to use PascalCasing for methods. Having it all lowercase feels out of place. We can't use `SHA1` as that conflicts with the type in the `System.Security.Cryptography` namespace. How about naming it `ComputeSHA1`?

```suggestion
			`private static string ComputeSHA1(string input) {
```